src/components/academy/Missions.tsx

```diff
@@ -1,29 +1,59 @@
 import { Button, Card, Icon, Intent, NonIdealState, Spinner, Text } from '@blueprintjs/core'
 import { IconNames } from '@blueprintjs/icons'
 import * as React from 'react'
+import { RouteComponentProps } from 'react-router'
+import { NavLink } from 'react-router-dom'
+
+import AssessmentContainer, {
+  OwnProps as AssessmentProps
+} from '../../containers/AssessmentContainer'
 import ContentDisplay, { IContentDisplayProps } from '../commons/ContentDisplay'
 
 export type MissionInfo = {
+  id: number
   title: string
   description: string
 }
 
-export interface IMissionsProps {
+export interface IMissionParams {
+  missionId?: string
+}
+
+export interface IMissionsProps extends RouteComponentProps<IMissionParams> {
   missionsInfo?: MissionInfo[]
   handleMissionsInfoFetch: () => void
 }
 
+export type StateProps = Pick<IMissionsProps, 'missionsInfo'>
+export type DispatchProps = Pick<IMissionsProps, 'handleMissionsInfoFetch'>
+
 class Missions extends React.Component<IMissionsProps, {}> {
   public render() {
-    const props: IContentDisplayProps = {
-      display: <MissionInfoCard missionsInfo={this.props.missionsInfo} />,
-      loadContentDispatch: this.props.handleMissionsInfoFetch
+    // make missionIdParam a number
+    let missionIdParam: number | null =
+      this.props.match.params.missionId === undefined
+        ? NaN
+        : parseInt(this.props.match.params.missionId, 10)
+    // set as null if the parsing failed
+    missionIdParam = Number.isInteger(missionIdParam) ? missionIdParam : null
+
+    // if there is no mission specified, Render only information.
+    if (missionIdParam === null) {
```

Comment:
I was considering between this and using a `Switch`. However, I think using `Switch` would cause the parsing of the missionId to be pushed to a later stage, making it more confusing. Here, it is parsed immediately (i.e the first place it can be accessed) and thereafter we do not need to worry about it being a string.